docs/API.md

```diff
@@ -437,8 +411,82 @@ Options
 There's a general method of setting options on a socket in the SRT C API, similar
 to the system setsockopt/getsockopt functions.
 
-Synopsis
---------
+Types used in socket options
+----------------------------
+
+Possible types of options are:
+
+* `int32_t` - usually you can treat it as `int` equivalent, as this type
+wasn't changed the size on 64-bit systems, although the definition is using
+`int32_t` type so that the bit size is fixed by definition. In some cases
+the value is expressed using an enumeration type (see below).
+
+* `int64_t` - some options need the paramter specified as 64-bit integer
+
+* `bool` - You need to use a boolean type, the builtin one in C++ and the
+one from `<stdbool.h>` for C, although when setting an option, passing the
+value through `int` type is also tolerated and properly recognized. When
+getting an option, however, you better use the `bool` type, although you
+can risk passing a variable of `int` type initialized with 0 and then check if
+the resulting value is equal to 0 (just don't compare the result with 1).
+
+* `string` - When setting, you pass the character array pointer as value
+and the string length as length. When getting, you have to pass a long
+enough array (with specified size in the size variable). Every option with
+this type that can be read should specify the maximum length of that array.
```

Comment:
```suggestion
* `int32_t` - This type does not change size on 64-bit systems, and can usually be 
treated as an `int` equivalent. The bit size is fixed by definition. In some cases the 
value is expressed using an enumeration type (see below).

* `int64_t` - Some options need the parameter specified as a 64-bit integer

* `bool` - Requires the use of a boolean type (`<stdbool.h>` for C, or built-in for C++).
When setting an option, passing the value through an `int` type is also properly recognized. When getting an option, however, you should use the `bool` type, although you
can risk passing a variable of `int` type initialized with 0 and then check if 
the resulting value is equal to 0 (just don't compare the result with 1).

* `string` - When setting an option, pass the character array pointer as value and 
the string length as length. When getting, pass an array of sufficient size (as specified 
in the size variable). Every option with this type that can be read should specify the 
maximum length of that array.
```

Reply:
As for `int32_t`: what I meant here is that the size of `int` type didn't change in 64-bit systems, meaning, you can still use `int` instead of `int32_t` - just `int32_t`, as a fixed-size integer, is used for clarity. I changed it to:

```
* `int32_t` - this type can usually be treated as an `int` equivalent since
this type does not change size on 64-bit systems. For clarity, options use
this fixed size integer. In some cases the value is expressed using an
enumeration type (see below).
```

Reply:
Very good.